projects/step-frontend/src/lib/modules/timeseries/modules/legacy/components/time-series-dashboard/time-series-dashboard.component.ts

```diff
@@ -27,9 +27,17 @@ import { FilterBarComponent } from '../../../filter-bar';
   imports: [COMMON_IMPORTS, FilterBarComponent, ChartsViewComponent],
 })
 export class TimeSeriesDashboardComponent implements OnInit, OnDestroy {
+  @Input()
+  set timeRange(value: TimeRangePickerSelection) {
+    if (value) {
+      this.handleTimeRangeChange({ selection: value, triggerRefresh: true });
+    }
+  }
```

Comment:
@lukasz-lepa 
It's better to avoid input setters.
Please use `ngOnChanges` hook or `input signal + effect` instead

Reply:
Changed for ngOnChanges